### ISSUE TITLE
Current requests cannot be canceled in Polaris

### ISSUE THREAD

**Issue by danfox01** [baltimorecounty/BCPL-assets#399]
Returns a a generic error message: An error was encountered.

According to @jdomasky this functionality has worked in the past. It's unclear if this was tested since the upgrade, though it is part of the test plan.

**Comment by mxsnyder** [comment]
This is `/polaris/PatronAccount/requests.aspx`.

**Comment by danfox01** [comment]
Steps to recreate:
1. Log in to [Polaris](https://catalog.bcpl.lib.md.us/polaris/)
1. Select "[Requests](https://catalog.bcpl.lib.md.us/polaris/patronaccount/requests.aspx)" on the left nav.
1. Select a checkbox next to one of your active requests (Karrie will need to add these to your test account if you don't have any.)
1. Choose "Cancel Selected"
1. Choose "Submit"

Instead of cancelling the request, an error message is displayed and the request is not cancelled.

**Comment by mxsnyder** [comment]
Here is the error from the server:

```
Event code: 3005 
Event message: An unhandled exception has occurred. 
Event time: 4/25/2018 10:15:13 PM 
Event time (UTC): 4/26/2018 2:15:13 AM 
Event ID: eb2af6d321204b70ad020a27a1a4aa86 
Event sequence: 39 
Event occurrence: 1 
Event detail code: 0 

Application information: 
   Application domain: /LM/W3SVC/1/ROOT/Polaris-2-131691799287361826 
   Trust level: Full 
   Application Virtual Path: /Polaris 
   Application Path: C:\Program Files\Polaris\5.5\PowerPAC\ 
   Machine name: ILS-TEST 

Process information: 
   Process ID: 7988 
   Process name: w3wp.exe 
   Account name: NT AUTHORITY\NETWORK SERVICE 

Exception information: 
   Exception type: FormatException 
   Exception message: Input string was not in a correct format.
  at System.Number.StringToNumber(String str, NumberStyles options, NumberBuffer& number, NumberFormatInfo info, Boolean parseDecimal)
  at System.Number.ParseInt32(String s, NumberStyles style, NumberFormatInfo info)
  at System.Convert.ToInt32(String value)
  at PowerPAC.patronaccount.Requests.linkbutton_Click(Object source, CommandEventArgs e)
  at System.Web.UI.WebControls.Button.OnCommand(CommandEventArgs e)
  at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)



Request information: 
   Request URL: https://ils-test.bcpl.lib.md.us:443/polaris/patronaccount/requests.aspx 
   Request path: /polaris/patronaccount/requests.aspx 
   User host address: 172.28.1.204 
   User:  
   Is authenticated: False 
   Authentication Type:  
   Thread account name: NT AUTHORITY\NETWORK SERVICE 

Thread information: 
   Thread ID: 11 
   Thread account name: NT AUTHORITY\NETWORK SERVICE 
   Is impersonating: False 
   Stack trace:    at System.Number.StringToNumber(String str, NumberStyles options, NumberBuffer& number, NumberFormatInfo info, Boolean parseDecimal)
  at System.Number.ParseInt32(String s, NumberStyles style, NumberFormatInfo info)
  at System.Convert.ToInt32(String value)
  at PowerPAC.patronaccount.Requests.linkbutton_Click(Object source, CommandEventArgs e)
```

**Comment by martypowell** [comment]
Status Update:

The last version of this working has been moved to production, so this is no longer a problem. However, we lost the tablenator (don't worry it'll be baacccckkkkk).

@danfox01 tablenator is in ils-test, let me know if you are able to cancel and active item. If so , we can move this as a hotfix.

**Comment by danfox01** [comment]
@martypowell the version in ils-test is not working again -- I get the error when trying to cancel.

The version that is in ils-test is tablenated **and** has the modified column order. Is that what you intended? Do we want to say if the tablenated version with the original column order will solve the problem?

**Comment by martypowell** [comment]
@danfox01 that is actually good, this means tablenator is breaking the cancel. I will look into this.

fyi @mxsnyder

**Comment by martypowell** [comment]
@danfox01 can you please try this again in ils-test, it should have tablenator, and hoping you are also able to cancel.

**Comment by danfox01** [comment]
@martypowell,  there is no tablenator in ils-test

**Comment by martypowell** [comment]
@danfox01 yea we messed up...working on getting that in now.

**Comment by martypowell** [comment]
@danfox01 this is back on ils-test

**Comment by danfox01** [comment]
Interesting... So, it works in desktop view, but once it gets tablenated at mobile, I still get an error.

**Comment by martypowell** [comment]
that makes total sense actually, although now the solution is more murky.

**Comment by martypowell** [comment]
@danfox01 ready for testing.

just need to hear back from you on whether we want the table with the content container around it, or without it, like on requests

**Comment by martypowell** [comment]
@danfox01 in dev, pending code review

**Comment by danfox01** [comment]
@martypowell, two showstoppers reported by @jdomasky:
- Tablenator doesn't 'nate in IE11
- On the Requests page, when sized down to mobile size in Chrome and FF, the Title and Hold Position turn blank.

Let me know if you'd prefer to see separate issues opened for these.

**Comment by martypowell** [comment]
@danfox01 ready for review in dev, I don't have any hold position records in an account, well need to test that somehow.

**Comment by jdomasky** [comment]
I tested canceling a request in Polaris dev and I was successful.
Also, I have a request with a hold position and I confirmed that the hold position remained visible at mobile widths in desktop Firefox and on an Android phone.
And finally, I sized down IE 11 to mobile width and the tablenator kicked in.
Based on my tests, it seems like this issue is fixed in dev, including both of yesterday's showstoppers.
